src/mmw/apps/modeling/geoprocessing.py

```diff
@@ -139,19 +141,25 @@ def histogram_finish(job_id, retry):
     """
     This is the bottom-half of the function.
     """
-    def dict_to_array(d):
-        result = []
-        for k, v in d.iteritems():
-            [k1, k2] = map(int, re.sub('[^0-9,]', '', k).split(','))
-            result.append(((k1, k2), v))
-        return result
-
     host = settings.GEOP['host']
     port = settings.GEOP['port']
 
-    data = sjs_retrieve(host, port, job_id, retry)
+    sjs_result = sjs_retrieve(host, port, job_id, retry)
+    data = {make_tuple(key[4:]): val for key, val in sjs_result.items()}
+
+    # Map NODATA soil cells cells to 3
+    # This was previously done within the SummaryJob: https://github.com/WikiWatershed/mmw-geoprocessing/blob/0d95dee35e729d9fd2f58fb9e73a69dcbe61df61/summary/src/main/scala/SummaryJob.scala#L106  # NOQA
+    # but since MapshedJob doesn't support remapping, we do it here.
+    nlcd_soil_count = {}
+    for (n, s), count in data.iteritems():
+        s2 = s if s != settings.NODATA else 3
+        nlcd_soil_count[(n, s2)] = count + nlcd_soil_count.get((n, s2), 0)
+
+    result = []
+    for (n, s), count in nlcd_soil_count.iteritems():
```

Comment:
This is the 3rd iteration over (mostly) the same data set.  I was skeptical at first why they shouldn't just be processed in a single loop, but given the small number of elements, it seems a little more clarity is more important than efficiency.  I'd suggest a small comment to each step describing the transformation process.